internal/component/database_observability/mysql/collector/explain_plan.go

```diff
@@ -562,19 +581,27 @@ func (c *ExplainPlan) populateQueryCache(ctx context.Context) error {
 	defer rs.Close()
 
 	// Populate cache
+SCHEMA_DENYLIST:
 	for rs.Next() {
 		if err := rs.Err(); err != nil {
 			level.Error(c.logger).Log("msg", "failed to iterate rs digests for explain plans", "err", err)
 			return err
 		}
 
-		var qi queryInfo
+		qi := &queryInfo{failureCount: 0}
 		var ls time.Time
 		if err = rs.Scan(&qi.schemaName, &qi.digest, &qi.queryText, &ls); err != nil {
 			level.Error(c.logger).Log("msg", "failed to scan digest for explain plans", "err", err)
 			return err
 		}
-		c.queryCache = append(c.queryCache, qi)
+		for _, schema := range c.schemaDenyList {
```

Comment:
I think you can use something like https://pkg.go.dev/slices#Contains

Reply:
Right, of course 🤦‍♂️